Marketplace/guitar/guitar.js

```diff
@@ -0,0 +1,666 @@
+                sounds["Yellow" + i + "R"] = SoundCache.getSound(Script.resolvePath("sounds/yellow/" + i + "R.wav"));
+                sounds["Green" + i + "L"] = SoundCache.getSound(Script.resolvePath("sounds/green/" + i + "L.wav"));
+                sounds["Green" + i + "R"] = SoundCache.getSound(Script.resolvePath("sounds/green/" + i + "R.wav"));
+                sounds["Blue" + i + "L"] = SoundCache.getSound(Script.resolvePath("sounds/blue/" + i + "L.wav"));
+                sounds["Blue" + i + "R"] = SoundCache.getSound(Script.resolvePath("sounds/blue/" + i + "R.wav"));
+            }
+        },
+        mousePress: function(id, event) {
+            if (overlays.indexOf(id) !== NOT_FOUND) {
+                if (timeout) {
+                    Script.clearTimeout(timeout);
+                    timeout = null;
+                }
+                timeout = Script.setTimeout(function(){
+                    _this.releaseGrab();
+                    timeout = null;
+                }, STILL_PLAYING_TIMEOUT_MS);
+                if (ampCheckInterval) {
+                    Script.clearTimeout(ampCheckInterval);
+                }
+                ampCheckInterval = Script.setInterval(this.updateAmpPositions, UPDATE_POSITION_MS);
+                var name = Overlays.getProperty(id, 'name');
+                if (name.indexOf("Neck") !== -1) {
+                    neckGrabColor = name.substr(NECK_COLOR_INDEX);
+                } else {
+                    var currentSoundL = sounds[neckGrabColor + name.substr(RIFF_NUMBER_INDEX) + "L"];
+                    var currentSoundR = sounds[neckGrabColor + name.substr(RIFF_NUMBER_INDEX) + "R"];
+                    if (currentSoundL.downloaded) {
+                        if (injectorL) {
+                            injectorL.stop();
+                        }
+                        injectorL = Audio.playSound(currentSoundL, {
+                            position: ampLeftPosition,
+                            volume: AUDIO_VOLUME_LEVEL,
+                            loop: true
+                        });
+                    }
+                    if (currentSoundR.downloaded) {
+                        if (injectorR) {
+                            injectorR.stop();
+                        }
+                        injectorR = Audio.playSound(currentSoundR, {
+                            position: ampRightPosition,
+                            volume: AUDIO_VOLUME_LEVEL,
+                            loop: true
+                        });
+                    }
+                }
+            }
+        },
+        setNeckColor: function(buttonID, params) {
+            neckGrabColor = params[0];
+        },
+        startNearGrab: function() {
+            if (!colorsShowing) {
+                this.createFingertipEntity("LeftHandMiddle4");
+                this.createFingertipEntity("RightHandMiddle4");
+                if (interval) {
```

Comment:
What is the purpose of manually updating the fingertip entities in the updatePositions function? If they're attached to the avatar and don't have any kind of collisions with the avatar and full damping, I _believe_ we'd expect them to remain in place.

Reply:
The problem was that, due to the avatar moving while the initial fingertip positions are calculated, the entities would not be in the correct place at the end of each finger at first, so I updated the position every 100ms for 3 seconds total. I have not seen any problems with the position after this process, but maybe the checks can occur less often or over a shorter period?